etna/analysis/outliers/isolation_forest_outliers.py

```diff
@@ -0,0 +1,174 @@
+
+    Parameters
+    ----------
+    ts:
+        TSDataset with timeseries data
+    in_column:
+        Name of the column in which the anomaly is searching
+    features_to_use:
+        List of feature column names to use for anomaly detection
+    features_to_ignore:
+        List of feature column names to exclude from anomaly detection
+    ignore_missing:
+        Whether to ignore missing values inside a series
+    n_estimators:
+        The number of base estimators in the ensemble
+    max_samples:
+        The number of samples to draw from X to train each base estimator
+            *  If int, then draw max_samples samples.
+
+            *  If float, then draw max_samples * X.shape[0] samples.
+
+            *  If “auto”, then max_samples=min(256, n_samples).
+
+        If max_samples is larger than the number of samples provided, all samples will be used for all trees (no sampling).
+    contamination:
+        The amount of contamination of the data set, i.e. the proportion of outliers in the data set.
+        Used when fitting to define the threshold on the scores of the samples.
+            *  If ‘auto’, the threshold is determined as in the original paper.
+
+            *  If float, the contamination should be in the range (0, 0.5].
+    max_features:
+        The number of features to draw from X to train each base estimator.
+            *  If int, then draw max_features features.
+
+            *  If float, then draw `max(1, int(max_features * n_features_in_))` features.
+        Note: using a float number less than 1.0 or integer less than number of features
+        will enable feature subsampling and leads to a longer runtime.
+    bootstrap:
+            *  If True, individual trees are fit on random subsets of the training data sampled with replacement.
+            *  If False, sampling without replacement is performed.
+    n_jobs:
+        The number of jobs to run in parallel for both fit and predict.
+            *  None means 1 unless in a joblib.parallel_backend context.
+            *  -1 means using all processors
+    random_state:
+        Controls the pseudo-randomness of the selection of the feature and split values for
+        each branching step and each tree in the forest.
+    verbose:
+        Controls the verbosity of the tree building process.
+    warm_start:
+        When set to True, reuse the solution of the previous call to fit and add more estimators to the ensemble,
+        otherwise, just fit a whole new forest. See the Glossary.
+    index_only:
+        whether to return only outliers indices. If `False` will return outliers series
+
+    Returns
+    -------
+    :
+        dict of outliers in format {segment: [outliers_timestamps]}
+    """
+    df = _select_features(
+        ts=ts, in_column=in_column, features_to_use=features_to_use, features_to_ignore=features_to_ignore
+    )
+    model = IsolationForest(
```

Comment:
If we make separate instance per segment this parameter won't work, now it might be set with some results, by default it doesn't share the state between fits 
<img width="716" alt="Снимок экрана 2024-06-04 в 12 09 45" src="https://github.com/etna-team/etna/assets/55380696/f8ae77bb-e7f2-4f77-a19c-29342c228108">

Reply:
I see. So, if we use this parameter, detection results might depend on segment order in the dataset. This doesn't seem right. I think we should set `warm_start=False` and prevent users from setting it.